src/iop/hlrecovery.c

```diff
@@ -126,7 +152,7 @@ static float calc_weight(const float *s, const char *lmask, const size_t loc, co
   return smoothness * sval;
 }
 
-static void prepare_smooth_singles(char *lmask, float * restrict src, const float * restrict ref, const size_t width, const size_t height, const float clipval)
+static void prepare_smooth_singles(char *lmask, float * restrict src, const float * restrict ref, const int width, const int height, const float clipval)
```

Comment:
I think we should keep `size_t` here (see below plane_size which takes size_t as parameters).

Reply:
yes. Changed at two more places where also possible